homeassistant/components/geniushub/switch.py

```diff
@@ -0,0 +1,51 @@
+"""Support for Genius Hub switch/outlet devices."""
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from . import DOMAIN, GeniusZone
+
+GH_ZONE_ATTR = "on / off"
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType, config: ConfigType, async_add_entities, discovery_info=None
+) -> None:
+    """Set up the Genius Hub switch entities."""
+    if discovery_info is None:
+        return
+
+    broker = hass.data[DOMAIN]["broker"]
+
+    switches = [
+        GeniusSwitch(broker, z)
+        for z in broker.client.zone_objs
+        if GH_ZONE_ATTR in z.data["type"]
+    ]
+
+    async_add_entities(switches, update_before_add=True)
+
+
+class GeniusSwitch(GeniusZone, SwitchDevice):
```

Comment:
Unfortunately, you can't just inherit from `GeniusZone` - it is geared up for Climate/WaterHeater entities (e.g. it has `elf._max_temp = self._min_temp = self._supported_features = None`).

You'll have to split `GeniusZone` into two, or extend `GeniusDevice`...

Reply:
I see what you mean, on the other hand, they all seem to be zones in geniushub land (at least that's how my install works..) how about we have GeniusZone as a base, and GeniusZoneHeating / GeniusZoneOutlet ?

Reply:
Hmm... I don't think so - it adds two more classes, and another level of inheritance.

Reply:
Ok, you da boss (and python definitely isnt my forte)

Reply:
Actually, I just changed my mind (I do like a bit of inheritance)!

You're right - spilt GeniusZone in two: GeniusZone  & GeniusHeatingZone.

Then: 
 - GeniusHeatingZone, GeniusSwitch inherit from GeniusZone
 - GeniusClimate, GeniusWaterHeater inherit from GeniusHeatingZone

Reply:
Just after I uploaded the other one.. ;)

Reply:
.. redone..